### ISSUE TITLE
Format input sheets for families of hypothesis networks

### ISSUE THREAD

**Issue by kdahlquist** [kdahlquist/GRNmap#166]
@GraceJohnson , @khorstmann , and @tessaam are going to finish formatting their families of input files for next week.  Specific tasks include:

1. alphabatize gene names for each input workbook
2. add in degradation and production rates from Belle et al.: instructions found here: http://www.openwetware.org/wiki/BIOL398-04/S15:Week_13#Create_the_Input_Excel_Workbook_for_the_Model under bullets 3 and 4.
3. for the missing values in the expression sheets, we will do the following:
     - fill in with yellow, the cells with the missing values
     - fill in the missing values with the average of the remaining values
     - paste special, paste values to wipe out the equation and leave the number
4. Do all 6 strains even though eventually we will run only 5 due to a bug #163 .
5. Be careful that all the numbers are formatted as numbers in your worksheets.

@bklein7 will work with @khorstmann and @tessaam 
@maggie-oneil will work with @GraceJohnson

**Comment by GraceJohnson** [comment]
@kdahlquist, I checked the GitHub wiki page titled "How to format the input file for GRNmap", but I wanted to confirm that the instructions are correctly standardized on the beta code before I start formatting.

**Comment by kdahlquist** [comment]
You need to use the beta instructions found here:  https://github.com/kdahlquist/GRNmap/wiki/How-to-format-the-input-file-for-GRNmap-%28beta%29

I have verified that this formatting works on the beta code.

**Comment by kdahlquist** [comment]
These questions came in via e-mail from @tessaam.  See my answers directly below each question.

1. If there isn't any data in any of the four replicates for one of the time points, should we delete that gene? For example, if there isn't any data for any of the four 15 time points for a gene in one of the log2 expression sheets, should we delete that gene for the network?

Answer: you should either delete that gene, or alternately, you could consider deleting that strain's expression data and keeping that gene (you could try both).

2. For my network where we did not add in CIN5, GLN3, HAP4, HMO1, SWI4, and ZAP1, my network only had ZAP1 and did not have GLN3 which is the deletion strain I am focusing on. Should I add in GLN3, or focus my attention to the network where  CIN5, GLN3, HAP4, HMO1, SWI4, and ZAP1 were added. 

Answer:  don't add GLN3 to that family of networks if it was not there originally; that is what the other family is for.  Only use the strain data that are appropriate for that particular network.  For example, if HAP4 is not part of the network, then you shouldn't use the expression data for that strain.

**Comment by khorstmann** [comment]
Update on progress for today's meeting since I won't be there: 
-Was only able to edit/create 3 networks. 34-gene with the deletion strains, 33 gene without deletion strains, and 29 gene without deletion strains. 
-Not sure how my networks jumped from 33 to 29 genes in the family w/o deletion strains being added in purposefully, think there must've been a lot of insignificant p-values and/or zeros when we were cleaning them out in October. But checked the rest of networks I made and seems like most of them only decrease by around 1 gene per network.
- I was getting errors when trying to find the averages for the blank cells for some sheets. Noticed that it was because there was an error on some cells saying that the cells were words, not numbers. I think what may have happened was when copying over from either Microsoft Access or another sheet, it was copied over as a string and not as values. Easy fix but tedious, am still in the process of fixing those cells. 

Here are my three networks created: 
[ONLY_DNA_binding_dZAP1_29_genes_1_22_16.xlsx](https://github.com/kdahlquist/GRNmap/files/101310/ONLY_DNA_binding_dZAP1_29_genes_1_22_16.xlsx)
[ONLY_DNA_binding_dZAP1_34_genes_1_22_16.xlsx](https://github.com/kdahlquist/GRNmap/files/101309/ONLY_DNA_binding_dZAP1_34_genes_1_22_16.xlsx)
[with_deletions_ONLY_DNA_dZAP1_34_Genes_1_20_16.xlsx](https://github.com/kdahlquist/GRNmap/files/101308/with_deletions_ONLY_DNA_dZAP1_34_Genes_1_20_16.xlsx)

**Comment by GraceJohnson** [comment]
dHAP4 network family formatted according to instructions for the beta code. @maggie-oneil formatted the no deletion strain added subfamily (12 networks) while I formatted the deletion strain added subfamily (20 genes). Sample files from each subfamily were tested for correct formatting using the beta code (files successfully run, although no output files were created due to the unresolved issue #168 - Add functionality to turn L-curve analysis on and off).

**Comment by tessaam** [comment]
I have 51 different networks created. They are formatted for master, rather than beta. I am now in the process of correctly formatting the input sheets.
For the family of networks where CIN5, GLN3, HAP4, HMO1, SWI4, and ZAP1 were added there were two genes that did not have data in any of the four replicates for one of the time points. I created 17 pared down input sheets where the log2 expression sheets were deleted and 17 where the genes were deleted. 
For the family of networks where CIN5, GLN3, HAP4, HMO1, SWI4, and ZAP1 were not added, I also created 17 pared down input sheets.

**Comment by kdahlquist** [comment]
There are three more changes that are required for the optimization_parameters worksheet (they are minor):

We can now get rid of the row that says "Deletion".  The code now can figure out the gene that is deleted from the strain information.

We need to change the word "Model" to "production_function" (cell A8)

We need to add a row beneath "production_function" called "L_curve".  A zero value for this parameter means no L-curve analysis is done and a 1 value for this parameter means that an L-curve analysis will be run.

Our  next step will be to run an actual real L-curve analysis on our networks--I'll create another issue for this.

I will document these changes on the wiki page: https://github.com/kdahlquist/GRNmap/wiki/How-to-format-the-input-file-for-GRNmap-%28beta%29

**Comment by kdahlquist** [comment]
As documented in issue #176 we need to conform to a standard naming convention for naming our input sheets.

I would like everyone to go through their familes of input sheets and name them according to this convention.

I think that once this is done, I'd like to collect everyone's input sheets again so I have everyone's data.  @khorstmann is going to give me her collection already so we can troubleshoot issues with her L-curve plots #172.  

Once they have been renamed and given to @kdahlquist, we can close this issue.

**Comment by kdahlquist** [comment]
This has been completed for the Spring 2016 semester; compiling and organizing these files for @kdahlquist is part of issue #202 .
